python/rpdk/java/templates/init/guided_aws/StubCreateHandler.java

```diff
@@ -0,0 +1,138 @@
+                    .stabilize(this::stabilizedOnCreate)
+                    .progress())
+
+            // STEP 3 [TODO: describe call/chain to return the resource model]
+            .then(progress -> new ReadHandler().handleRequest(proxy, request, callbackContext, proxyClient, logger));
+    }
+
+    /**
+     * If your service API does not distinguish duplicate create requests against some identifier (e.g; resource Name)
+     * and instead returns a 200 even though a resource already exists, you must first check if the resource exists here
+     * NOTE: If your service API throws 'ResourceAlreadyExistsException' for create requests this method is not necessary
+     * @param proxy Amazon webservice proxy to inject credentials correctly.
+     * @param request incoming resource handler request
+     * @param progressEvent event of the previous state indicating success, in progress with delay callback or failed state
+     * @return progressEvent indicating success, in progress with delay callback or failed state
+     */
+    private ProgressEvent<ResourceModel, CallbackContext> checkForPreCreateResourceExistence(
+        final AmazonWebServicesClientProxy proxy,
+        final ResourceHandlerRequest<ResourceModel> request,
+        final ProgressEvent<ResourceModel, CallbackContext> progressEvent) {
+        final ResourceModel model = progressEvent.getResourceModel();
+        final CallbackContext callbackContext = progressEvent.getCallbackContext();
+        try {
+            // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/ReadHandler.java#L40-L46
+            new ReadHandler().handleRequest(proxy, request, callbackContext, logger);
+            throw new CfnAlreadyExistsException(ResourceModel.TYPE_NAME, Objects.toString(model.getPrimaryIdentifier()));
+        } catch (CfnNotFoundException e) {
+            logger.log(model.getPrimaryIdentifier() + " does not exist; creating the resource.");
+            return ProgressEvent.progress(model, callbackContext);
+        }
+    }
+
+    /**
+     * Implement client invocation of the create request through the proxyClient, which is already initialised with
+     * caller credentials, correct region and retry settings
+     * @param awsRequest the aws service request to create a resource
+     * @param proxyClient the aws service client to make the call
+     * @return awsResponse create resource response
+     */
+    private AwsResponse createResource(
+        final AwsRequest awsRequest,
+        final ProxyClient<SdkClient> proxyClient) {
+        AwsResponse awsResponse = null;
+        try {
+            // TODO: add custom create resource logic
+            // e.g. https://github.com/aws-cloudformation/aws-cloudformation-resource-providers-logs/blob/master/aws-logs-loggroup/src/main/java/software/amazon/logs/loggroup/CreateHandler.java#L37-L43
```

Comment:
I wouldn't call this "custom create resource logic"... all we are doing is calling create.  Surely there is a better example?

This is a good example of an API that throws AlreadyExists exception though. I would link here in `step 1 [check if resource already exists]`